rough-work-shreyan/qs_5.Rmd

```diff
@@ -0,0 +1,131 @@
+---
+title: "PBSR ASSIGNMENT 2"
+author: "SHREYAN CHAKRABORTY"
+date: "`r Sys.Date()`"
+output:
+  html_document: default
+  pdf_document: default
+---
+
+```{r setup, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+```
+
+
+
+```{r}
+library(tidyverse)
+library(quantmod)
+getSymbols('TCS.NS')
+tail(TCS.NS)
+
+```
+
+```{r}
+plot(TCS.NS$TCS.NS.Adjusted)
+```
+```{r}
+getSymbols('^NSEI')
+tail(NSEI)
+```
+```{r}
+plot(NSEI$NSEI.Adjusted)
+```
+```{r}
+TCS_rt = diff(log(TCS.NS$TCS.NS.Adjusted))
+Nifty_rt = diff(log(NSEI$NSEI.Adjusted))
+retrn = cbind.xts(TCS_rt,Nifty_rt)
+retrn = na.omit(data.frame(retrn))
+plot(retrn$NSEI.Adjusted,retrn$TCS.NS.Adjusted
+     ,pch=20
+     ,xlab='Market Return'
+     ,ylab='TCS Return'
+     ,xlim=c(-0.18,0.18)
+     ,ylim=c(-0.18,0.18))
+grid(col='grey',lty=1)
+
+```
+
+where $\mathbb{E}(\varepsilon)=0$ and $\mathbb{V}ar(\varepsilon)=\sigma^2$.
+
+1. Estimate the parameters of the models $\theta=(\alpha,\beta,\sigma)$ using the method of moments type plug-in estimator discussed in the class.
+
+2. Estimate the parameters using the `lm` built-in function of `R`. Note that `lm` using the OLS method.
+
+
+
+```{r}
+#For method of moment estimation, the equations are as follows:
+#E(TCS)=alpha +beta*E(Nifty)
+#E(Nifty(TCS-alpha-beta*Nifty))=0
+#E(TCS-alpha-beta*Nifty)^2=sigma^2
+attach(retrn)
+tail(retrn)
+
+#creating a 2X2 coefficient matrix for solving of system of equations:
+x<-array(0,4)
+x[1] = 1
+x[2] <- mean(NSEI.Adjusted)
+x[3] <- mean(NSEI.Adjusted)
+x[4] <- mean((NSEI.Adjusted)*(NSEI.Adjusted))
+A <- matrix(x,nrow=2)
+
+#inserting another column containing TCS*NSE value named as "C"
+retrn <- retrn %>% 
+    mutate(C=TCS.NS.Adjusted*NSEI.Adjusted)
+
+#storing values in 2X1 matrix for solving of system of equations:
+y <- array(0,2)
+y[1] <- mean(TCS.NS.Adjusted)
+y[2] <- mean(C)
```

Comment:
you cant call C like that as it has not been attached. Either do `attach(retrn)` before this line or call `retrn$C`